src/services/pasteEdits.ts

```diff
@@ -77,7 +76,11 @@ function pasteEdits(
         if (copiedFrom?.range) {
             Debug.assert(copiedFrom.range.length === pastedText.length);
             copiedFrom.range.forEach(copy => {
-                addRange(statements, copiedFrom.file.statements, getLineOfLocalPosition(copiedFrom.file, copy.pos), getLineOfLocalPosition(copiedFrom.file, copy.end) + 1);
+                const statementsInSourceFile = copiedFrom.file.statements;
+                const startNodeIndex = findIndex(statementsInSourceFile, s => s.end > copy.pos);
+                if (startNodeIndex === -1) return undefined;
+                const endNodeIndex = findIndex(statementsInSourceFile, s => s.end >= copy.end, startNodeIndex);

```

Comment:
This includes the next statement if the end of the selection includes any leading trivia of the next statement:

```ts
// both statements get added to `statements`:
[|console.log(a);
|]
console.log(b);
```